src/OpenTelemetry.AutoInstrumentation.Native/cor_profiler.cpp

```diff
@@ -241,8 +241,7 @@ HRESULT STDMETHODCALLTYPE CorProfiler::Initialize(IUnknown* cor_profiler_info_un
         opcodes_names.push_back("->");      // CEE_SWITCH_ARG
     }
 
-    //
-    managed_profiler_assembly_reference = AssemblyReference::GetFromCache(managed_profiler_full_assembly_version);
+    managed_profiler_assembly_reference = AssemblyReference::GetFromCache(GetBytecodeInstrumentationAssembly());
```

Comment:
Changes based on https://github.com/DataDog/dd-trace-dotnet/pull/750

Without this, OpenTelemetry.AutoInstrumentation was loaded to non domain-neutral.

@zacharycmontoya, fyi.

Reply:
ah, this is because we have two different "full assembly version" strings: one with the strong signature and another without it.